jina_commons/indexers/dump.py

```diff
@@ -107,18 +108,61 @@ def _write_shard_files(
     ids_fh.write(id_ + '\n')
 
 
-def import_vectors(path: str, pea_id: str):
+def _import_vectors_psql(path, shard_ids):
+    # see https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING
+    connection_string = urllib.parse.urlparse(path)
+    if connection_string.hostname is None or \
+            connection_string.port is None or \
+            connection_string.username is None or \
+            connection_string.password is None or \
+            connection_string.path is None or \
+            'table=' not in connection_string.query:
+        raise ValueError(
+            'Missing one required parameter in postgres path. Must be like: "postgres://username:password@host:port/dbname?table=table_name"')
+    connection = connect(
+        host=connection_string.hostname,
+        port=connection_string.port,
+        user=connection_string.username,
+        password=connection_string.password,
+        dbname=connection_string.path.strip('/')
+    )
+    cursor = connection.cursor()
+    cursor.execute(
+        f'SELECT ID, DOC FROM {connection_string.query.split("table=")[1]} WHERE SHARD IN (",".join(shard_ids))'
```

Comment:
This is to tightly coupled to the actual PSQLIndexer. I would prefer, that the PSQLIndexer defines a dump function on the PSQL database and this function is called here.